#             Data Analysis of my Favourite Data Science channel of all time.





### WHY THIS PROJECT?

It all started with the urge of learning Machine Learning and I was unable to understand it in an efficient manner because I didn't have a strong technical background but scrolling down and down on YouTube finding a perfect solutions for my doubts, I found out a channel called CampusX. It was an amaxing channel by an Indian Teacher, The way he taught machine learning was very unique and very intuitive. I later scrolled more in his videos and entered the comment session and saw that there are many comments saying that the teaching is amazing but the Teacher as well his Channel was very very underrated.
So I wanted to do a data analysis for his channel comparing it to other 50 such YouTube channels.


### HOW THE DATA IS COLLECTED?
There is an extension for browsers called as 'TubeBuddy', it offers few numbers about the YouTube channel as we start any of their video or view their channel page.
Like this Data of 51 such DataScience channel was collected and later the anaylsis was performed. The whole analysis is comparing 'CampusX' with 50 other channels.

In [1]:
import numpy as np
import pandas as pd

In [2]:
df = pd.read_csv('data-science-youtube-channel-comparison.csv')

In [3]:
df.head()

channel name  views   subs  last 30 days views  num videos  \
0          Data School   9.73  193.0               106.0         138   
1        Bhavesh Bhatt   2.72   37.4                72.7         292   
2  Unfold Data Science   1.72   27.9               123.0         369   
3      Abhishek Thakur   1.74   69.8                59.6         139   
4            AppliedAI  13.20   77.9                98.6         497   

      how old  
0  01/05/2014  
1  12/11/2017  
2  02/08/2019  
3  26/12/2019  
4  28/09/2017

In [4]:
df.shape

(51, 6)

### **Here we can see that in total we have 51 data science channels and for all the channels we have 6 features representing analytics of that channels.**

## The 6 features are : 

1.) **channel name** : Name of the channel(str)

2.) **views** : The total number of views on that channel upto '2022-01-28' in(int)

3.) **subs** : Number of Subscribers(int)

4.) **last 30 days views** : Aggregate of views on all his videos for the last 30 days(int)

5.) **num videos** : Total number of videos in his/her channel(int)

6.) **how old** : When was the channel created(str). This data contains the date on which he/she uploaded his/her first video.

In [5]:
df['views'] = df['views'] * 1000000
df['subs'] = df['subs'] * 1000
df['last 30 days views'] = df['last 30 days views']*1000

In [6]:
df.head()

channel name       views      subs  last 30 days views  num videos  \
0          Data School   9730000.0  193000.0            106000.0         138   
1        Bhavesh Bhatt   2720000.0   37400.0             72700.0         292   
2  Unfold Data Science   1720000.0   27900.0            123000.0         369   
3      Abhishek Thakur   1740000.0   69800.0             59600.0         139   
4            AppliedAI  13200000.0   77900.0             98600.0         497   

      how old  
0  01/05/2014  
1  12/11/2017  
2  02/08/2019  
3  26/12/2019  
4  28/09/2017

In [7]:
df['how old'] = pd.to_datetime(df['how old'])
basedate = pd.Timestamp('2022-01-28')
df['channel_age'] = (basedate-df['how old']).dt.days

In [8]:
df.head()

channel name       views      subs  last 30 days views  num videos  \
0          Data School   9730000.0  193000.0            106000.0         138   
1        Bhavesh Bhatt   2720000.0   37400.0             72700.0         292   
2  Unfold Data Science   1720000.0   27900.0            123000.0         369   
3      Abhishek Thakur   1740000.0   69800.0             59600.0         139   
4            AppliedAI  13200000.0   77900.0             98600.0         497   

     how old  channel_age  
0 2014-01-05         2945  
1 2017-12-11         1509  
2 2019-02-08         1085  
3 2019-12-26          764  
4 2017-09-28         1583

In [9]:
df.drop('how old',axis=1,inplace=True)

In [10]:
df.head()

channel name       views      subs  last 30 days views  num videos  \
0          Data School   9730000.0  193000.0            106000.0         138   
1        Bhavesh Bhatt   2720000.0   37400.0             72700.0         292   
2  Unfold Data Science   1720000.0   27900.0            123000.0         369   
3      Abhishek Thakur   1740000.0   69800.0             59600.0         139   
4            AppliedAI  13200000.0   77900.0             98600.0         497   

   channel_age  
0         2945  
1         1509  
2         1085  
3          764  
4         1583

In [11]:
df.rename(columns={'channel name':'channel_name','last 30 days views':'last_30_days_views','num videos':'total_no_of_videos'},inplace=True)

In [12]:
df.sample(5)

channel_name       views      subs  last_30_days_views  \
38           Tina Huang  10800000.0  294000.0            651000.0   
9             Statquest  35900000.0  702000.0           1390000.0   
50               Jovian    950000.0   27000.0             59700.0   
16            KGPTalkie   4450000.0   39800.0             88800.0   
2   Unfold Data Science   1720000.0   27900.0            123000.0   

    total_no_of_videos  channel_age  
38                  92          555  
9                  217         2524  
50                 148          813  
16                 593         1352  
2                  369         1085

In [13]:
df.isnull().sum()

channel_name          0
views                 0
subs                  1
last_30_days_views    0
total_no_of_videos    0
channel_age           0
dtype: int64

In [14]:
df[df['subs'].isna()]

channel_name      views  subs  last_30_days_views  \
30  Abhishek Agarrwal  7430000.0   NaN            196000.0   

    total_no_of_videos  channel_age  
30                1190         1920

In [15]:
from sklearn.impute import KNNImputer
knn = KNNImputer(n_neighbors=3,weights='distance')
knn.fit_transform(df.iloc[:,1:])[30]

array([7.43000000e+06, 1.14611478e+05, 1.96000000e+05, 1.19000000e+03,
       1.92000000e+03])

In [16]:
df['subs'].fillna(114611,inplace=True)

In [17]:
df.isnull().sum()

channel_name          0
views                 0
subs                  0
last_30_days_views    0
total_no_of_videos    0
channel_age           0
dtype: int64

In [18]:
df

channel_name        views       subs  last_30_days_views  \
0                 Data School    9730000.0   193000.0            106000.0   
1               Bhavesh Bhatt    2720000.0    37400.0             72700.0   
2         Unfold Data Science    1720000.0    27900.0            123000.0   
3             Abhishek Thakur    1740000.0    69800.0             59600.0   
4                   AppliedAI   13200000.0    77900.0             98600.0   
5       5 minutes engineering   37900000.0   303000.0           1610000.0   
6             Serrano Academy    4770000.0    96400.0             70200.0   
7                  Krish Naik   51100000.0   556000.0           1650000.0   
8                  Codebasics   35000000.0   557000.0           1470000.0   
9                   Statquest   35900000.0   702000.0           1390000.0   
10                    CampusX    1500000.0    18000.0            198000.0   
11          Satyajit Pattnaik     480000.0    19400.0             56400.0   
12              Daniel Bourke    5280000.0   112000.0             84900.0   
13                 Deeplizard    9330000.0   117000.0            167000.0   
14                    Ken Jee    6060000.0   195000.0            153000.0   
15            Python Engineer    6960000.0   115000.0            823000.0   
16                  KGPTalkie    4450000.0    39800.0             88800.0   
17             AI Engineering    1950000.0    62200.0             61400.0   
18            Nicolas Renotte    3440000.0    67900.0            312000.0   
19                 Siddaharan    1020000.0    48600.0            118000.0   
20       Indian AI Production    3530000.0    31900.0            119000.0   
21          The AI University    1330000.0    24000.0             46200.0   
22              Harshit Tyagi     180000.0     8030.0              5240.0   
23           Developer Ashish     320000.0    16200.0             13300.0   
24            Normalized Nerd    1300000.0    30600.0            117000.0   
25               Smitha Kolan    1090000.0    37800.0             45400.0   
26         Shashank Kalanithi    3260000.0    81900.0            202000.0   
27              DigitalSreeni    2570000.0    38400.0            198000.0   
28             Brandon Rohrer    4390000.0    77300.0             46500.0   
29                  Ranji Raj    6210000.0    42300.0            116000.0   
30          Abhishek Agarrwal    7430000.0   114611.0            196000.0   
31            Deeplearning.ai   11300000.0   170000.0            181000.0   
32     Deep Learning in Hindi     330000.0     4470.0              8410.0   
33              Fahad Hussain    1310000.0    16100.0             46500.0   
34           Data Science Jay     970000.0    22400.0             66600.0   
35              Sundas Khalid    2150000.0    64700.0            174000.0   
36           Seattle Data Guy    1200000.0    26000.0            108000.0   
37             Data Professor    2790000.0   116000.0            130000.0   
38                 Tina Huang   10800000.0   294000.0            651000.0   
39  Data Science Career Coach    1540000.0    28000.0            124000.0   
40                    Sentdex  101000000.0  1110000.0            588000.0   
41          Data Science Dojo    4690000.0    82700.0             66500.0   
42     Data Science Tutorials    3890000.0    38100.0            130000.0   
43              Brandon Foltz   23200000.0   262000.0            179000.0   
44                   Datacamp   20800000.0   129000.0            627000.0   
45               The Semiclon    2170000.0    23500.0             20300.0   
46                 Ritvikmath    5780000.0    79900.0            215000.0   
47                     kaggle    2940000.0   103000.0             55600.0   
48                    ineuron    1130000.0    41700.0            705000.0   
49                Springboard    3900000.0    53000.0            141000.0   
50                     Jovian     950000.0    27000.0             59700.0   

    total_no

***Here onwards I am going to compare CampusX with other 50 channels using 7 Metrics***

### View Metirc

In [19]:
# top 10 YouTube channels according to views.
df.sort_values('views',ascending=False)[['channel_name','views']].head(10)

channel_name        views
40                Sentdex  101000000.0
7              Krish Naik   51100000.0
5   5 minutes engineering   37900000.0
9               Statquest   35900000.0
8              Codebasics   35000000.0
43          Brandon Foltz   23200000.0
44               Datacamp   20800000.0
4               AppliedAI   13200000.0
31        Deeplearning.ai   11300000.0
38             Tina Huang   10800000.0

In [20]:
# View ranking of CampusX
print(np.where(df.sort_values('views',ascending=False)['channel_name'] == 'CampusX')[0][0] ,'out of 51')

37 out of 51


In [21]:
# Percentile ranking of CampusX
print(round((14/51)*100),'percentile')

27 percentile


### Subs Metric

In [22]:
# top 10 YouTube channels according to Subs.
df.sort_values('subs',ascending = False)[['channel_name','subs']].head(10)

channel_name       subs
40                Sentdex  1110000.0
9               Statquest   702000.0
8              Codebasics   557000.0
7              Krish Naik   556000.0
5   5 minutes engineering   303000.0
38             Tina Huang   294000.0
43          Brandon Foltz   262000.0
14                Ken Jee   195000.0
0             Data School   193000.0
31        Deeplearning.ai   170000.0

In [23]:
# Subs ranking of CampusX
print(np.where(df.sort_values('subs',ascending=False)['channel_name'] == 'CampusX')[0][0],'out of 51')

46 out of 51


In [24]:
# Percentile ranking of CampusX
print(round((5/51)*100),'percentile')

10 percentile


### Number of Videos Metric

In [25]:
# top 10 YouTube channels according to Total Number of videos on their channels.
df.sort_values('total_no_of_videos',ascending = False)[['channel_name','total_no_of_videos']].head(10)

channel_name  total_no_of_videos
39  Data Science Career Coach                1410
5       5 minutes engineering                1390
7                  Krish Naik                1360
40                    Sentdex                1250
44                   Datacamp                1240
30          Abhishek Agarrwal                1190
10                    CampusX                1100
42     Data Science Tutorials                 869
33              Fahad Hussain                 741
16                  KGPTalkie                 593

In [26]:
#Total number of videos ranking of CampusX
print(np.where(df.sort_values('total_no_of_videos',ascending=False)['channel_name']=='CampusX')[0][0],'out of 51')

6 out of 51


In [27]:
# Percentile ranking of CampusX
print(round((45/51)*100),'percentile')

88 percentile


### Channel Age Metric

In [28]:
# top 10 YouTube channels according to the channel age(how old the channel is)
old_df = df.sort_values('channel_age',ascending=False)[['channel_name','channel_age']].head(10)
old_df['channel_age'] = round(old_df['channel_age']/356)
old_df

channel_name  channel_age
43    Brandon Foltz         10.0
40          Sentdex          9.0
0       Data School          8.0
9         Statquest          7.0
46       Ritvikmath          7.0
44         Datacamp          7.0
8        Codebasics          6.0
28   Brandon Rohrer          6.0
29        Ranji Raj          6.0
6   Serrano Academy          6.0

In [29]:
# Channel Age ranking of CampusX
print(np.where(df.sort_values('channel_age',ascending=False)['channel_name']=='CampusX')[0][0],'out of 51')

38 out of 51


In [30]:
# Percentile ranking of campusx
print(round((13/51)*100),'percentile')

25 percentile


### 30 days views Metric

In [31]:
# top 10 YouTube channels according to views achieved in the last 30 days from 2022-01-28
df.sort_values('last_30_days_views',ascending=False)[['channel_name','last_30_days_views']].head(10)

channel_name  last_30_days_views
7              Krish Naik           1650000.0
5   5 minutes engineering           1610000.0
8              Codebasics           1470000.0
9               Statquest           1390000.0
15        Python Engineer            823000.0
48                ineuron            705000.0
38             Tina Huang            651000.0
44               Datacamp            627000.0
40                Sentdex            588000.0
18        Nicolas Renotte            312000.0

In [32]:
# Last 30 days views of CampusX
print(np.where(df.sort_values('last_30_days_views',ascending=False)['channel_name']=='CampusX')[0][0],'out of 51')

13 out of 51


In [33]:
# Percentile ranking of CampusX
print(round((38/51)*100),'percentile')

75 percentile


### Views/Sub count Metric

In [34]:
# top 10 YouTube channels according to views/sub count
df['views/sub count'] = round(df['views'] / df['subs'])

In [35]:
df.head()

channel_name       views      subs  last_30_days_views  \
0          Data School   9730000.0  193000.0            106000.0   
1        Bhavesh Bhatt   2720000.0   37400.0             72700.0   
2  Unfold Data Science   1720000.0   27900.0            123000.0   
3      Abhishek Thakur   1740000.0   69800.0             59600.0   
4            AppliedAI  13200000.0   77900.0             98600.0   

   total_no_of_videos  channel_age  views/sub count  
0                 138         2945             50.0  
1                 292         1509             73.0  
2                 369         1085             62.0  
3                 139          764             25.0  
4                 497         1583            169.0

In [36]:
# top 10 channels which require very less view for subsribe
df.sort_values('views/sub count',ascending = True)[['channel_name','views/sub count']].head(10)

channel_name  views/sub count
23   Developer Ashish             20.0
19         Siddaharan             21.0
22      Harshit Tyagi             22.0
37     Data Professor             24.0
3     Abhishek Thakur             25.0
11  Satyajit Pattnaik             25.0
48            ineuron             27.0
25       Smitha Kolan             29.0
47             kaggle             29.0
17     AI Engineering             31.0

In [37]:
df[df.sort_values('views/sub count',ascending = True)['channel_name'] == 'CampusX']

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  """Entry point for launching an IPython kernel.


channel_name      views     subs  last_30_days_views  total_no_of_videos  \
10      CampusX  1500000.0  18000.0            198000.0                1100   

    channel_age  views/sub count  
10          859             83.0

Conclusion : It wil take 83 views for 1 person to subscribe.

Or put it in other laymen interpretations, out of every 83 viewers only 1 will subscribe.

In [38]:
# Views/Sub count ranking of CampusX
print(np.where(df.sort_values('views/sub count',ascending=True)['channel_name']=='CampusX')[0][0],'out of 51')

39 out of 51


In [39]:
#percentile ranking of CampusX
print(round((12/51)*100),'percentile')

24 percentile


### Video/Sub ratio Metric

In [40]:
# creating videos/sub ratio column in df
df['videos/sub ratio'] = df['total_no_of_videos'] / df['subs']

In [41]:
df.head()

channel_name       views      subs  last_30_days_views  \
0          Data School   9730000.0  193000.0            106000.0   
1        Bhavesh Bhatt   2720000.0   37400.0             72700.0   
2  Unfold Data Science   1720000.0   27900.0            123000.0   
3      Abhishek Thakur   1740000.0   69800.0             59600.0   
4            AppliedAI  13200000.0   77900.0             98600.0   

   total_no_of_videos  channel_age  views/sub count  videos/sub ratio  
0                 138         2945             50.0          0.000715  
1                 292         1509             73.0          0.007807  
2                 369         1085             62.0          0.013226  
3                 139          764             25.0          0.001991  
4                 497         1583            169.0          0.006380

In [42]:
# top 10 YouTube channels according to videos/sub ratio
df.sort_values('videos/sub ratio',ascending=True)[['channel_name','videos/sub ratio']].head(10)

channel_name  videos/sub ratio
9         Statquest          0.000309
38       Tina Huang          0.000313
6   Serrano Academy          0.000446
0       Data School          0.000715
35    Sundas Khalid          0.000788
8        Codebasics          0.000885
43    Brandon Foltz          0.000989
40          Sentdex          0.001126
14          Ken Jee          0.001179
31  Deeplearning.ai          0.001288

In [43]:
df[df.sort_values('videos/sub ratio',ascending=True)[['channel_name','videos/sub ratio']]['channel_name'] == 'CampusX']

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.
  """Entry point for launching an IPython kernel.


channel_name      views     subs  last_30_days_views  total_no_of_videos  \
10      CampusX  1500000.0  18000.0            198000.0                1100   

    channel_age  views/sub count  videos/sub ratio  
10          859             83.0          0.061111

Conclusion : It requires 0.061111 video for one person to subscribe

In [44]:
# Videos/Sub ratio of CampusX
print(np.where(df.sort_values('videos/sub ratio',ascending=True)['channel_name']=='CampusX')[0][0],'out of 51')

50 out of 51


In [45]:
# Percentile ranking of CampusX
print(round((1/51)*100),'percentile')

2 percentile


### Visualization of views VS subs for all 51 channels

In [46]:
import plotly.express as px

In [47]:
fig = px.scatter(data_frame=df,x=df['views'],y=df['subs'],hover_name=df['channel_name'])
fig.show()